applications/StructuralMechanicsApplication/custom_strategies/custom_strategies/formfinding_updated_reference_strategy.hpp

```diff
@@ -835,6 +215,95 @@ namespace Kratos
         ///@name Protected Operators
         ///@{
 
+void UpdateDatabase(
+        TSystemMatrixType& A,
+        TSystemVectorType& Dx,
+        TSystemVectorType& b,
+        const bool MoveMesh
+    ) override
+    {
+        if(mIncludeLineSearch == false){
+            BaseType::UpdateDatabase(A,Dx, b, MoveMesh);
+        }
+        else{
+            typename TSchemeType::Pointer pScheme = this->GetScheme();
+            typename TBuilderAndSolverType::Pointer pBuilderAndSolver = this->GetBuilderAndSolver();
+
+            TSystemVectorType aux(b.size()); //TODO: do it by using the space
+            TSparseSpace::Assign(aux,0.5, Dx);
+
+            //compute residual without update
+            TSparseSpace::SetToZero(b);
+            pBuilderAndSolver->BuildRHS(pScheme, BaseType::GetModelPart(), b );
+            double ro = TSparseSpace::TwoNorm(b);
+
+            //compute half step residual
+            BaseType::UpdateDatabase(A,aux,b,MoveMesh);
+            TSparseSpace::SetToZero(b);
+            std::map<int,Matrix> prestress;
```

Comment:
use unordered map pls, this should be faster for what you are using it

Reply:
Indeed

Reply:
done